apps/src/applab/applab.js

```diff
@@ -262,6 +262,45 @@ function renderFooterInSharedGame() {
   );
 }
 
+Applab.autogenerateML = function() {
+  $.ajax({
+    url: '/api/v1/ml_models',
+    method: 'GET'
+  }).then(modelData => {
+    console.log('modelData', modelData);
+    modelData.selectedFeatures.forEach(feature => {
+      var label = designMode.createElement('LABEL');
+      label.textContent = feature + ':';
+      label.id = feature + '_label';
+      if (Object.keys(modelData.featureNumberKey).includes(feature)) {
+        var select = designMode.createElement('DROPDOWN');
+        select.id = feature + '_dropdown';
+        // App Lab automatically addss "option 1" and "option 2", remove them.
+        select.options.remove(0);
+        select.options.remove(0);
+        Object.keys(modelData.featureNumberKey[feature]).forEach(option => {
+          var optionElement = document.createElement('option');
+          optionElement.text = option;
+          select.options.add(optionElement);
+        });
+      } else {
+        var input = designMode.createElement('TEXT_INPUT');
+        input.id = feature + '_input';
+      }
+    });
+    var label = designMode.createElement('LABEL');
+    label.textContent = modelData.labelColumn;
+    // TODO: this could be problematic if the name isn't formatted appropriately
+    label.id = modelData.name + '_label';
+    var prediction = designMode.createElement('TEXT_INPUT');
+    prediction.text = '?';
+    prediction.id = modelData.name + '_prediction';
+    var predictButton = designMode.createElement('BUTTON');
+    predictButton.textContent = 'Predict';
+    predictButton.id = modelData.name + '_predict';
+  });
+};
```

Comment:
The good news: 
This code correctly retrieves the user's most recently saved machine learning model, and automatically generates the correct design elements based on the selected features of that model and plops them into the design mode visualizer. 

The bad news: 
1.) the design elements are re-created each time the app loads, so this code is obviously not being called in the right place/at the right time. 
2.) the elements are all stacked on each other in the top left corner; we'll have to find a way to cleverly arrange them. 

In addition to the 2 hurdles above, the next step is to generate the code that should accompany these design elements (getting the values from dropdowns etc). My lead so far on that front is `Applab.appendToEditor(code)`, which coincidentally is what libraries use to generate blocks for the imported functions. 

We'll also want to hide all of this behind and experiment flag and set up some trigger for when App Lab should auto-generate and when it shouldn't. To extend the query param idea, I think we could set it up similarly to `enableMaker` where when `enableAI` (or whatever the flag is) is on we'd do both the autogenerating and set up an additional section of the block toolbox with the AI blocks.

Reply:
updates: 

1.) I now set and check a localStorage variable to prevent the code and design elements from duplicating on page reload. 

2.) `designMode.createElement` takes X, Y parameters to set the element's starting position. Leveraging those parameters + a little math, makes a less cluttered layout without any overlaps.  Positioning will still be problematic if there are more generated elements than will fit on a single screen; but that feels like a problem for future polishing. 

3.) I put the autogen code behind the experiment flag already in use `applab-ml`; once we have a clearer plan for how users will selected models in App Lab, we can revisit if we'd like an additional param or other mechanisms for triggering element and code autogeneration.

Reply:
re `1.)` There is a risk that a student will change computer and return to the same app lab level, and won't have the same local storage, and will get the same code auto-generated again.  I think we should rely on something more server-side than client-side to make sure we auto-generate only once.

To capture some of our offline discussion, there are a few proposed principles for auto-generation: 
- app lab should be responsible for modifying its level content, rather than ai lab
- ai lab shouldn't be able to force an export to app lab because it might force an overwrite of a student's previous work

To that end, I'd suggest a model in which ai lab writes an identifier for the level (set via levelbuilder) to `UserMlModel`.  App lab levels can then be tagged (in levelbuilder) to auto-generate using the most-recently-saved model with that same identifier.  We'd use something similar to the "starter code" system to make sure that app lab only does this the first time, even if the student changes computer.

Some benefits of this system:
- an app lab level can auto-generate with a model that was saved on any earlier level.
- app lab can give helpful directions if there isn't a model saved with the appropriate identifier yet.
- ai lab is agnostic about app lab's existence.

Reply:
re:
> the design elements are re-created each time the app loads, so this code is obviously not being called in the right place/at the right time

unless i'm misunderstanding what this method does, i think all of this code should be a react component like the other design elements in this directory (they all have very similar structure): https://github.com/code-dot-org/code-dot-org/tree/staging/apps/src/applab/designElements